### PR TITLE
doc: Provide documentation for new contributors

### LINKED ISSUE #691: Developer documentation
#### What happened:
There is no documentation for developers on how to run our tests on different platforms or onboard a new platform to CI. It's there in the code (GH actions workflows, e2e tests), but not summarized except in a generic way in CONTRIBUTING.md.

#### What you expected to happen:
A document telling a developer:
 - how to run the operator locally or using a local cluster with modifications
 - how to run the e2e tests or a subset of the tests
 - what to add and where if a new platform is to be supported

We could use the Machine Config Operator's hacking guide as inspiration: https://github.com/openshift/machine-api-operator/blob/master/docs/dev/hacking-guide.md

### PR DESCRIPTION
**Pull request by jhrozek**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
New contributors might have a hard time building SPO themselves or
running tests. This patch adds a document that walks through building
SPO locally, running and adding unit tests and running e2e tests
locally.

#### Which issue(s) this PR fixes:
Fixes #691

#### Does this PR have test?
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: hacking.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+be used locally, but rather deployed in a cluster from pre-built images,
+it's often useful to be able to run a quick build or a test on your local machine.
+
+Depending on what OS and version you are developing on and what features you
+want to build with, you might need to either install extra dependencies or
+disable features that would require them.
+
+There are currently two optional features at build time
+- eBPF based recording
+  - This feature requires a rather new `libelf` and `libbpf` versions
```
Comment (saschagrunert) [inline hacking.md:17]: To be fully complete, libbpf depends on libelf which depends on libz

File: hacking.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+## Building SPO locally
+Even though SPO is a Kubernetes operator and as such is normally not meant to
+be used locally, but rather deployed in a cluster from pre-built images,
+it's often useful to be able to run a quick build or a test on your local machine.
+
+Depending on what OS and version you are developing on and what features you
+want to build with, you might need to either install extra dependencies or
+disable features that would require them.
+
+There are currently two optional features at build time
```
Comment (saschagrunert) [inline hacking.md:15]: libseccomp is a non-optional dependency which is required for building the operator, too. I mean, yeah we disable it for Darwin build environments, but it truly degrades the bpf recorder and enricher feature.

**Review by saschagrunert — COMMENTED** [review]
Great docs, I just have two nits.

File: hacking.md (hunk 1 above)
Comment (jhrozek) [reply hacking.md:17]: ok, fixed (hopefully :-))

File: hacking.md (hunk 2 above)
Comment (jhrozek) [reply hacking.md:15]: sorry, but I'm not sure what do you want me to fix here? The doc describes that the eBPF recorder is optional, not seccomp.

Feel free to provide some proposal of wording that you think is better.

Comment (jhrozek) [reply hacking.md:15]: oh, do you want me to add that `libseccomp` is always required? Sorry, I'm a little slow today :-)

Comment (saschagrunert) [reply hacking.md:15]: Yeah but do we wanna mention that libseccomp is required for building the operator as well?

**Comment by saschagrunert** [comment]
Agreed and merged

**Pull request merged**